### ISSUE TITLE
Each epoch's validation done with 1 image only?

### ISSUE THREAD

**Issue by WeiChihChern** [qubvel/segmentation_models#396]
As title,

or does the validation performance metrics are calculated with entire validation set?

**Comment by JordanMakesMaps** [comment]
See Segmentation_models/losses.py

```python
Args:
        class_weights: Array (``np.array``) of class weights (``len(weights) = num_classes``).
        class_indexes: Optional integer or list of integers, classes to consider, if ``None`` all classes are used.
        per_image: If ``True`` loss is calculated for each image in batch and then averaged, # <------------------
            else loss is calculated for the whole batch.
        smooth: Value to avoid division by zero.
    Returns:
         A callable ``jaccard_loss`` instance. Can be used in ``model.compile(...)`` function
         or combined with other losses.

```
